config/initializers/rack/attack.rb

```diff
@@ -0,0 +1,7 @@
+class Rack::Attack
+  throttle("search_throttle", limit: 5, period: 1) do |request|
+    if request.path.starts_with?("/search/") && request.env["HTTP_FASTLY_CLIENT_IP"].present?
```

Comment:
Nice, the second condition should cover the case of other communities not having shielding enabled, if it's optional

Reply:
Yeah, eventually I would love for Fastly to be optional since it really isn't necessary and is an optimization so I figured I would take the safe route. Even better, but I didn't do here would be too store the request.ip in the event that the fastly variable is not present. TODO!